lib/sinon/stub.js

```diff
@@ -196,6 +196,15 @@ var proto = {
 
     onThirdCall: function onThirdCall() {
         return this.onCall(2);
+    },
+
+    withArgs: function withArgs() {
+        var fake = spy.withArgs.apply(this, arguments);
```

Comment:
Since this is being moved to another file, we have the opportunity to also rename the variable from `fake` (which can easily be confused with `fake`) into something that is unambiguous.

Perhaps `spy` or `instance`?

Reply:
Hmmm, the `spy.withArgs` implementation calls it a `fake` and there is the `this.fakes` array where the newly created `fake` is being kept. I think it should either have this name or we should remove the ambiguity from the codebase by renaming it everywhere in `spy.js` and stub.js`.

Reply:
Fair enough, we can rename them all in a later PR ... all at the same time

Reply:
Go ahead and merge this